x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/AutodetectBuilder.java

```diff
@@ -249,6 +251,7 @@ private void buildScheduledEventsConfig(List<String> command) throws IOException
     }
 
     private void buildJobConfig(List<String> command) throws IOException {
+        Files.createDirectories(env.tmpFile());
```

Comment:
This simple fix resolves the problem locally.

I don't know if my local environment is close enough to production w.r.t. security, permissions, etc, to judge whether this will also work in production.

Reply:
Update: this fix works in a Dockerized Ubuntu 20.04.

The following flow succeeds:
- start Elasticsearch
- start job
- stop a job
- remove tmp dir
- start a job

The deleted tmp dir is successfully recreated.

Reply:
Great, thanks for testing that. The final thing to confirm is the permissions.

Reply:
Note: without this fix you get
```
java.nio.file.NoSuchFileException: /tmp/elasticsearch-14675733582508699252/config6396271037101692016.json
```